bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/AbstractTextEditor.java

```diff
@@ -7095,6 +7114,38 @@ protected void doRestoreState(IMemento memento) {
 			textWidget.setHorizontalPixel(horizontalPixel.intValue());
 	}
 
+	@SuppressWarnings("boxing")
+	private ISelection readSelectionFromMemento(IMemento memento) {
+		if (memento.getBoolean(TAG_IS_MULTISELECTION)) {
```

Comment:
```suggestion
		if (memento.getInteger(TAG_SELECTION_OFFSET + "0") != null) {
```
and them something like
```java
int i = 0;
Integer offset = null;
while ((offset = memento.getInteger(TAG_SELECTION_OFFSET + i)) != null) {
	Integer length = memento.getInteger(TAG_SELECTION_LENGTH + i);
	if (length == null) {
		continue;
	}
	regions.add(new Region(offset, length));
}
```